FSW/src/util/dataCollection.cpp

```diff
@@ -256,4 +262,61 @@ unsigned long calcTimestamp() {
     unsigned long timestamp = currentTimeRelative; 
 
     return timestamp;
+}
+
+/* - - - - - - downlink - - - - - - *
+ *  
+ * Usage:
+ *  downlinks all files on the flash module
+ * 
+ * Inputs:
+ *  none
+ *  
+ * Outputs:
+ *  none
+ */
+void downlink() {
+    
+    if (newDownlink) { // reset asynchronous variables if it is a new downlink
+        filename[FILE_IDX_OFFSET] = '0';
+        downlinkFileIdx = 0;
+        downlinkFileCount = 0;
+        newDownlink = false;
+    }
+
+    if (downlinkFileIdx < MAXFILES){ // check all science possible filenames
```

Comment:
I really like that you've staggered the downlink over consecutive loop runthroughs.